package/MDAnalysis/analysis/contacts.py

```diff
@@ -445,10 +481,8 @@ def __init__(self, u, select, refgroup, method="hard_cut", radius=4.5,
         self.initial_contacts = []
 
         #get dimension of box if pbc set to True
-        if self.pbc:
-            self._get_box = lambda ts: ts.dimensions
-        else:
-            self._get_box = lambda ts: None
+        self.pbc = pbc
+        self._get_box = functools.partial(get_box, pbc=self.pbc)
```

Comment:
```suggestion
        self._get_box = lambda ts: ts.dimensions if self.pbc else None
```
?

Reply:
@RMeli I tried running it with the suggestion locally and I get

`AttributeError: Can't get local object 'Contacts.__init__.<locals>.<lambda>'`

It is somehow due to the parallelization having problems with lamba, it also does not like this


```
        self._get_box = functools.partial(
            lambda ts, pbc: ts.dimensions if pbc else None, pbc=self.pbc
        )
```
it leads to the same error, so yes I would agree that the function is quite general, so moving 
`def get_box()` to be a function in the `Contacts` class would be fine? since that would
works with the tests

Reply:
I wouldn't put `get_box()` elsewhere because it's really a bit clunky and not really that useful except that it's needed here (or do we have other places where we need it??). I'd keep it as local and non-public as possible. Writing it as a lambda would be the nicest way but if that does not work (... good to know and a bit sad ...) then I'd make it a static private class method
```python
@staticmethod
def _get_box_func(ts, pbc):
    """... keep the written docs ... """
    return ts.dimensions if pbc else None
```
which you can then use for the partial.

Add a comment to `self._get_box = functools.partial(self._get_box_func, pbc=self.pbc)` why all of this is necessary.

----

(I assume that following will run into the same issue as lambdas:
```python
def get_box(ts, pbc=self.pbc):
   return ts.dimensions if pbc else None
self._get_box = get_box
```
as it's also local, right?)

Reply:
Thanks @talagayev for the explanation, I didn't know a lambda would not work in this case. I think defining a function instead of a lambda will in our in the same issue, as you suspect @orbeckst, but haven't tried. If we don't want to move the function elsewhere, I agree it should be private and not exposed by the module.

Reply:
Yes, tried it right now locally, leads sadly to the `Can't get local object 'Contacts.__init__.<locals>.get_box'`

Reply:
but yes making it more local static method is a more sophisticated way then it is currently, will adjust it then :)

Reply:
@orbeckst @RMeli I moved it now as you suggested and would re request the review, to see if it is correct now :)